### ISSUE TITLE
Fix argument splitting in compiletest

### ISSUE THREAD

**Issue by jieyouxu** [rust-lang/rust#132599]
We should fix how arguments are split in compiletest, this involves multiple places:

- The custom diff tool
- `//@ compile-flags` and such

They usually involve converting from a user-provided escaped string to `Command::new()` and `Command::arg` and such, but yeah. Command arg splitting in compiletest is currently very naive and usually just splits on whitespace or is otherwise very basic, we should probably just use a proper library to handle the splitting...

**Comment by ChrisDenton** [comment]
> we should probably just use a proper library to handle the splitting...

Definitely! I believe [`shlex`](https://docs.rs/shlex) would be a reasonable choice.
